optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -343,4 +386,60 @@ private String convertClassInfosToString(Collection<ClassInfo> classInfos) {
         }
     }
 
+    private OptaPlannerGizmoInfo generateDomainAccessors(IndexView indexView,
+            BuildProducer<GeneratedClassBuildItem> generatedClasses,
+            BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        Collection<AnnotationInstance> membersToGeneratedAccessorsFor = new ArrayList<>();
+        ClassOutput classOutput = new GeneratedClassGizmoAdaptor(generatedClasses, true);
+        ClassOutput debuggableClassOutput = (className, bytes) -> {
+            final String DEBUG_CLASSES_DIR = "target/optaplanner-generated-classes";
+            if (DEBUG_CLASSES_DIR != null) {
+                Path pathToFile = Paths.get(DEBUG_CLASSES_DIR, className.replace('.', '/') + ".class");
+                try {
+                    Files.createDirectories(pathToFile.getParent());
+                    Files.write(pathToFile, bytes);
+                } catch (IOException e) {
+                    System.out.println("Failed writing debug class: " + pathToFile);
+                    e.printStackTrace();
+                }
+            }
+            classOutput.write(className, bytes);
+        };
+
+        Map<String, java.lang.reflect.Type> gizmoMemberAccessorNameToGenericType = new HashMap<>();
+        Map<String, AnnotatedElement> gizmoMemberAccessorNameToAnnotatedElement = new HashMap<>();
+
+        GizmoMemberAccessor.setGizmoClassLoader(Thread.currentThread().getContextClassLoader(),
+                gizmoMemberAccessorNameToGenericType,
+                gizmoMemberAccessorNameToAnnotatedElement);
+        for (DotName dotName : DotNames.GIZMO_MEMBER_ACCESSOR_ANNOTATIONS) {
+            membersToGeneratedAccessorsFor.addAll(indexView.getAnnotations(dotName));
+        }
+        for (AnnotationInstance annotatedMember : membersToGeneratedAccessorsFor) {
+            switch (annotatedMember.target().kind()) {
+                case FIELD: {
+                    FieldInfo fieldInfo = annotatedMember.target().asField();
+                    ClassInfo classInfo = fieldInfo.declaringClass();
+                    GizmoMemberAccessorCreator.generateFieldAccessor(annotatedMember, indexView, debuggableClassOutput,
+                            classInfo,
+                            fieldInfo, transformers);
+                    break;
+                }
+                case METHOD: {
+                    MethodInfo methodInfo = annotatedMember.target().asMethod();
+                    ClassInfo classInfo = methodInfo.declaringClass();
+                    GizmoMemberAccessorCreator.generateMethodAccessor(annotatedMember, indexView, debuggableClassOutput,
+                            classInfo,
+                            methodInfo, transformers);
+                    break;
+                }
+                default: {
+                    throw new IllegalStateException(annotatedMember + " is not on a field or method.");
```

Comment:
error messages: put value's between braces.
```suggestion
                    throw new IllegalStateException("The member (" + annotatedMember + ") is not on a field or method.");
```